.github/workflows/build.yaml

```diff
@@ -0,0 +1,89 @@
+name: Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: ["contribute"]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        runner: [windows-viewer]
+        configuration: [ReleaseOS]
+        addrsize: [64]
+        include:
+          - runner: windows-viewer
+            configuration: ReleaseOS
+            addrsize: 32
+    runs-on: ${{ matrix.runner }}
+    env:
+      AUTOBUILD_CONFIGURATION: ${{ matrix.configuration }}
+      AUTOBUILD_ADDRSIZE: ${{ matrix.addrsize }}
+      AUTOBUILD_INSTALLABLE_CACHE: ${{ github.workspace }}/.autobuild-installables
+      AUTOBUILD_VARIABLES_FILE: ${{ github.workspace }}/.build-variables/variables
+      AUTOBUILD_VSVER: "170" # vs2k22
+      LOGFAIL: debug # Show details when tests fail
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Checkout build variables
+        uses: actions/checkout@v3
+        with:
+          repository: secondlife/build-variables
+          ref: viewer
+          path: .build-variables
+
+      - name: Install autobuild and python dependencies
+        run: pip3 install autobuild llbase
+
+      - name: Cache autobuild packages
```

Comment:
:1st_place_medal: Warms my heart to see dependency caching, and a nicely-thought-out cache key/restore keys!